src/etc/inc/system.inc

```diff
@@ -202,10 +202,11 @@ function system_resolvconf_generate($dynupdate = false) {
 				route_add_or_change("-host {$inet6}{$dnsserver} {$gatewayip}");
 			} else {
 				/* Remove old route when disable gw */
-				mwexec("/sbin/route -q delete -host {$inet6}{$dnsserver}");
+				$gatewayip = exec("/sbin/route -n show {$dnsserver} | grep gateway | cut -d' ' -f6");
```

Comment:
* Use `route -n get`, rather than `show` to be consistent with other uses elsewhere in our code.
* Use the full path to binaries.
* That `cut` is inaccurate. Swap both `grep` and `cut` out for a single `awk`:
```/sbin/route -n get {$dnsserver} | /usr/bin/awk '/gateway:/ {print $2;}'```

Reply:
done